### PR TITLE
Add refinery for honeycomb sampling

### PR DESCRIPTION
**Pull request by dwwoelfel**
Adds refinery to do honeycomb tail sampling for us.

Our configuration is set up to send all events that have an `error` property, and throttle all other events so that we send a max of about 30/second. That should keep us below our Honeycomb plan limits.

The refinery container sits next to the server container and we redirect all of our honeycomb API calls to the refinery container. Then the refinery container makes a decision on what to send based on the rules we defined in `server/refinery/rules.yaml`. More docs on those [here](https://docs.honeycomb.io/manage-data-volume/sample/honeycomb-refinery/sampling-methods/) and [here](https://github.com/honeycombio/refinery/blob/main/rules_complete.yaml).

We use a `EMAThroughputSampler` to set a target throughput for non-errors. It uses the `name` field to group spans for throttling, so that it doesn't completely drop certain spans. If we want to throttle certain spans more or with more granularity, we can add additional rules to the yaml file.

### CODE REVIEW

File: .github/workflows/clojure.yml [hunk 1]
```diff
@@ -146,10 +146,11 @@ jobs:
       working-directory: server
       run: clojure -M:lint "--config" ".clj-kondo/ci-config.edn"
 
-  publish:
+  publish-image:
```
Comment (dwwoelfel) [inline .github/workflows/clojure.yml:149]: Small change to make the deploys faster, we build the docker image while running the tests, but only create the eb application version if the tests pass.

File: server/docker-compose.yml [hunk 2]
```diff
@@ -22,3 +24,12 @@ services:
       - /var/log/eb-engine.log:/var/log/eb-engine.log:ro
       - /var/log/messages:/var/log/messages:ro
       - /var/log/eb-docker:/var/log/eb-docker:ro
+  refinery:
+    image: honeycombio/refinery:2.7.1
+    volumes:
+      - ./refinery/config.yaml:/etc/refinery/refinery.yaml
+      - ./refinery/rules.yaml:/etc/refinery/rules.yaml
+    environment:
+      - REFINERY_HONEYCOMB_API_KEY=${REFINERY_HONEYCOMB_API_KEY}
```
Comment (dwwoelfel) [inline server/docker-compose.yml:33]: I set the `REFINERY_HONEYCOMB_API_KEY` env var in the elastic beanstalk config.

File: server/src/instant/util/tracer.clj [hunk 3]
```diff
@@ -166,21 +167,21 @@
            (end-span! *span*))))))
 
 ;; (XXX)
-;; Given a `sample-rate`, we will randomly skip spans at that rate 
+;; Given a `sample-rate`, we will randomly skip spans at that rate
```
Comment (nezaj) [inline server/src/instant/util/tracer.clj:170]: Nit. Should we delete this comment now?

Comment (dwwoelfel) [reply server/src/instant/util/tracer.clj:170]: I think we should leave everything the same for a bit while we get comfortable with refinery, then come back and move any sampling configuration to the rules.yaml.

Comment (nezaj) [reply server/src/instant/util/tracer.clj:170]: Sounds good!

**Comment by stopachka** [comment]
Woohoo!!

**Review by nezaj — APPROVED** [review]

**Pull request merged**